### DISCUSSION TITLE
Mastodon UI: 4 images limit in any received post

### DISCUSSION THREAD

**Discussion by MarekBenjamin** [mastodon/mastodon#20149] [General]
Hi, I noticed when I receive post from #fediverse users using other software than #mastodon, like #friendica and especially #pixelfed they sometime include more than four images in their post. In the mastodon UI at max 4 images are shown from the original post are shown even if there were more images posted.

The problem: The selected four images shown in the mastodon ui seems to be completely random. Furthermore, and more sever: The is no single information that the original post contains more than four images. This is especially a problem IMHO since you can click through the four images seamlessly in the mastodon ui and therefore many uses might never discover the other available images by clicking on the link to the original post within another application. 

I have searched for a discussion about this issue but did not find one, so I started these. Pleas apologize if there is already a discussion / topic about that since I am completely new to github :-)

![Screenshot 2022-11-08 at 19-01-41 Friendica Social Network Marek Bachmann friendica @ Friendica Social Network](https://user-images.githubusercontent.com/117765478/200641179-67b2e18c-7d0c-4163-8238-a696003ca6ed.png)
![Screenshot 2022-11-08 190303](https://user-images.githubusercontent.com/117765478/200641300-289cc0e1-d7fd-4d39-859f-e68ff7237e83.png)

**Comment by trwnh** [comment]
#14336
